models/DataObject/Classificationstore.php

```diff
@@ -496,19 +510,38 @@ public function setGroupCollectionMapping($groupId = null, $collectionId = null)
      */
     public function getGroupCollectionMapping($groupId)
     {
-        return $this->groupCollectionMapping[$groupId] ?? null;
+        return $this->getGroupCollectionMappings()[$groupId] ?? null;
     }
 
     /**
      * @return Model\DataObject\Classificationstore\Group[]
      */
     public function getGroups(): array
     {
-        $activeGroups = $this->getActiveGroups();
+        $groups = Classificationstore::getActiveGroupsWithConfig($this);
+
+        $object = $this->getObject();
+        while (null !== ($parent = Service::hasInheritableParentObject($object))) {
+            $fieldDefintions = $parent->getClass()->getFieldDefinitions();
+            foreach ($fieldDefintions as $key => $fd) {
+                if ($fd instanceof Model\DataObject\ClassDefinition\Data\Classificationstore) {
+                    $getter = 'get' . ucfirst($key);
+                    $groups = array_merge(Classificationstore::getActiveGroupsWithConfig($parent->$getter()), $groups);
+                }
+            }
+            $object = $parent;
+        }
+
+        return $groups;
+    }
```

Comment:
Hi @robertSt7, 
I would suggest that this would be more readable
 
    ($parent = Service::hasInheritableParentObject($object)) !== null)
 
Also, as most of this code is same as in lines 470 - 484 what do you think about creating a private method and deciding with param for example how to build the final array to minimize redundancy? Something like

    private function getInheritedData(array $groups, bool $merge = false)
    {
        $object = $this->getObject();
        while (($parent = Service::hasInheritableParentObject($object)) !== null) {
            $fieldDefintions = $parent->getClass()->getFieldDefinitions();
            foreach ($fieldDefintions as $key => $fd) {
                if($fd instanceof Model\DataObject\ClassDefinition\Data\Classificationstore){
                    $getter = 'get' . ucfirst($key);
                    if ($merge) {
                        $groups = array_merge(Classificationstore::getActiveGroupsWithConfig($parent->$getter()), $groups);
                    } else {
                        $groups = $groups + $parent->$getter()->groupCollectionMapping;
                    }

                }
            }

            $object = $parent;
        }

        return $groups;
    }

Reply:
@lukmzig Done